source_code/src/FLASH/flash_mem.c

```diff
@@ -42,7 +42,9 @@
 void memoryBoundaryErrorCallback(void)
 {
     // We'll add more debug later if needed
+#ifndef MINI_BOOTLOADER
     usbPutstr("#MBE");
+#endif
```

Comment:
That us true, but the file is still compiled and linked. And because It cannot find the USB functions (which are NOT compiled, nor linked) the compiling will fail. No matter if the function is used.

We could get rid of this by compiling the USB code too and then the linker notices that this function is not required and does not use it. However this is a) useless and b) it pulls in the USB interrupt vectors of the USB core. And this gives additional overhead and also the bootloader should not use interrupts.

Thatswhy this define is essential.